js/src/views/Signer/store.js

```diff
@@ -49,8 +49,8 @@ export default class SignerStore {
   }
 
   @action unsubscribe () {
-    if (this._timeoutId) {
-      clearTimeout(this._timeoutId);
+    if (this.subscriptionId) {
```

Comment:
It should always wait for subscription id, otherwise you may end up with a dangling subscription.